src/create-route-map.js

```diff
@@ -70,6 +70,12 @@ function addRouteRecord (
         path || name
       )} cannot be a ` + `string id. Use an actual component instead.`
     )
+    if (route.component && route.component.toString) {
+      assert(
+        route.component.toString() !== '[object Promise]',
```

Comment:
Let's check for `then` in case of poly fills like at https://github.com/vuejs/vue-router-next/blob/master/src/navigationGuards.ts#L171